mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -240,30 +245,35 @@ async fn block_header_utxo_loop<T: UtxoCommonOps>(
             Err(e) => {
                 error!("Error {} on getting the height of the last stored header in DB!", e);
                 sync_status_loop_handle.notify_on_temp_error(e.to_string());
-                Timer::sleep(10.).await;
+                Timer::sleep(BLOCK_HEADERS_LOOP_SLEEP_TIMER).await;
                 continue;
             },
         };
 
-        // Todo: Add code for the case if a chain reorganization happens
-        if from_block_height == last_block_height {
-            sync_status_loop_handle.notify_sync_finished(last_block_height);
-            last_block_height = match coin.as_ref().rpc_client.get_block_count().compat().await {
+        let to_block_height = from_block_height + chunk_size;
+        let to_block_height = if to_block_height > block_count {
+            block_count
+        } else {
+            block_count = match coin.as_ref().rpc_client.get_block_count().compat().await {
                 Ok(h) => h,
                 Err(e) => {
                     error!("Error {} on getting the height of the latest block from rpc!", e);
                     sync_status_loop_handle.notify_on_temp_error(e.to_string());
-                    Timer::sleep(10.0).await;
+                    Timer::sleep(BLOCK_HEADERS_LOOP_SLEEP_TIMER).await;
                     continue;
                 },
             };
+            to_block_height
+        };
```

Comment:
This is not efficient. Let's see the following case:
`from_block_height = 1024, chunk_size = 10, block_count(local) = 1030, block_count(actual) = 1031`
`to_block_height = from_block_height + chunk_size = 1034`
Then `(to_block_height > block_count) = true` and `to_block_height := block_count(local) = 1030`.
In the end, we'll do 2 iterations to fetch all 1031 block headers.

Also let's see another inefficient case:
`from_block_height = 1000, chunk_size = 10, block_count(local) = block_count(actual) = 1030`
`to_block_height = from_block_height + chunk_size = 1034`
Then `(to_block_height > block_count) = false` and we'll fetch block_count despite the fact that at this moment we don't need to fetch it.

Reply:
Please consider these code changes https://github.com/KomodoPlatform/atomicDEX-API/commit/be6a0aa7ed8046f88ea914acb4d43aae7bdd06f6

Reply:
I haven't run this code, so please re-check me, I could make a mistake. Please also run locally and check if everything works as expected